src/bin/pregen.rs

```diff
@@ -81,6 +81,14 @@ fn main() {
         }
         None => find_patterns(all_targets),
     };
-
-    fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    if cli.only_tail {
+        fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    } else {
+        let mut pos_neg_values: BTreeMap<i64, (String, String)> = BTreeMap::new();
+        for (k, v) in all_data {

```

Comment:
Missing the zero check, does the resulting file have an entry for -0? Also would prefer more descriptive variable names here

Reply:
gonna test if rust is smart enough to -0==0

Reply:
yeah they are equivalent

Reply:
but it is a tiny bit of wasted processing power